### PR TITLE
Update to kirkstone.

### PR DESCRIPTION
**Pull request by perezmeyer**
In order to make the beaglebone-ai64 to work it is first required to switch to meta-ti's kirkstone branch, so we need to update all the relevant layers.

This also includes a fix for the gen_mod_headers script, as it seems Balena is passing a wrong 3rd argument. Fix by Jon Trulson <jtrulson@ics.com>

Changelog-entry: Update balena-beaglebone to kirkstone.

### CODE REVIEW

File: layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_5.4.bb [hunk 1]
```diff
@@ -25,14 +25,14 @@ TAG = "5.4.70-ti-r21"
...
-MACHINE_KERNEL_PR_append = "b"
+MACHINE_KERNEL_PR:append = "b"
 PR = "${MACHINE_KERNEL_PR}"
 
 KERNEL_CONFIG_DIR = "${S}/ti_config_fragments"
 
 # SoC specific config fragments are not enabled yet
-#KERNEL_CONFIG_FRAGMENTS_append_ti33x = " ${KERNEL_CONFIG_DIR}/am33xx_only.cfg"
-#KERNEL_CONFIG_FRAGMENTS_append_ti43x = " ${KERNEL_CONFIG_DIR}/am43xx_only.cfg"
+#KERNEL_CONFIG_FRAGMENTS:append_ti33x = " ${KERNEL_CONFIG_DIR}/am33xx_only.cfg"
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_5.4.bb:34]: I believe `_ti33x` also needs to be converted to the new syntax

File: layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend [hunk 2]
```diff
@@ -60,6 +60,6 @@ BALENA_CONFIGS[hm3301] = " \
 # still be aufs, so we need to include the aufs driver further
 # for them, as per the internal thread
 # https://www.flowdock.com/app/rulemotion/resin-devices/threads/K2TQiSUfNDqBT5Ih6cciNI2d9QJ
-BALENA_CONFIGS_append_beaglebone = " aufs"
-BALENA_CONFIGS_append_beaglebone-green = " aufs"
-BALENA_CONFIGS_append_beaglebone-green-wifi = " aufs"
+BALENA_CONFIGS:append_beaglebone = " aufs"
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend:63]: New syntax conversion here also is needed

File: layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend [hunk 3]
```diff
@@ -10,17 +10,17 @@ KERNEL_DEVICETREE_beaglebone += " \
 	am335x-bonegreen-gateway.dtb \
 "
 
-SRC_URI_append_beaglebone = " \
+SRC_URI:append_beaglebone = " \
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend:13]: New syntax conversion needed here

File: layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend [hunk 4]
```diff
@@ -10,17 +10,17 @@ KERNEL_DEVICETREE_beaglebone += " \
...
 	file://rtc-omap-Prevent-kernel-panic-and-reboot-on-shutdown.patch \
         file://0001-Revert-net-ethernet-ti-cpsw-fix-min-eth-packet-size.patch \
 "
 
-SRC_URI_append_beagleboard-xm = " \
+SRC_URI:append_beagleboard-xm = " \
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend:18]: New syntax conversion needed here

File: layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend (hunk 2 above)
Comment (perezmeyer) [reply layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend:63]: I'm actually on that!

File: layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend [hunk 5]
```diff
@@ -10,17 +10,17 @@ KERNEL_DEVICETREE_beaglebone += " \
...
 	file://0001-set-gpios-vaux3.patch \
 	file://0001-card-power-cycle.patch \
 "
 
-BALENA_CONFIGS_append_beagleboard-xm = " omap3_soc \
+BALENA_CONFIGS:append_beagleboard-xm = " omap3_soc \
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-kernel/linux/linux-beagleboard_%.bbappend:23]: New syntax conversion needed here

File: layers/meta-balena-beaglebone/recipes-core/images/balena-image.inc [hunk 6]
```diff
@@ -14,15 +14,15 @@ BALENA_BOOT_PARTITION_FILES_beagleboard-xm = " \
...
 
 # Beaglebone expects the kernel and dtbs on rootfs
-IMAGE_INSTALL_append = " kernel-image-initramfs kernel-devicetree wlconf"
+IMAGE_INSTALL:append = " kernel-image-initramfs kernel-devicetree wlconf"
 
-IMAGE_INSTALL_append = " beagle-serial"
+IMAGE_INSTALL:append = " beagle-serial"
 
-BALENA_BOOT_PARTITION_FILES_append_beaglebone-pocket = " uEnv.txt_internal:"
+BALENA_BOOT_PARTITION_FILES:append_beaglebone-pocket = " uEnv.txt_internal:"
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-core/images/balena-image.inc:21]: New syntax conversion needed here

File: layers/meta-balena-beaglebone/recipes-core/images/balena-image.inc [hunk 7]
```diff
@@ -1,6 +1,6 @@
-IMAGE_FSTYPES_append_beaglebone = " balenaos-img balenaos-img.bmap"
-IMAGE_FSTYPES_append_beaglebone-pocket = " balenaos-img"
-IMAGE_FSTYPES_append_beagleboard-xm = " balenaos-img"
+IMAGE_FSTYPES:append_beaglebone = " balenaos-img balenaos-img.bmap"
+IMAGE_FSTYPES:append_beaglebone-pocket = " balenaos-img"
+IMAGE_FSTYPES:append_beagleboard-xm = " balenaos-img"
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-core/images/balena-image.inc:3]: New syntax conversion needed here

File: layers/meta-balena-beaglebone/recipes-core/images/balena-image.bbappend [hunk 8]
```diff
@@ -1,4 +1,4 @@
 include balena-image.inc
 
-IMAGE_INSTALL_append_beaglebone = " bb-org-overlays fix-mmc-bbb bb-wl18xx-bluetooth bb-wl18xx-wlan0"
-IMAGE_INSTALL_append_beaglebone-pocket = " boot-scripts"
+IMAGE_INSTALL:append_beaglebone = " bb-org-overlays fix-mmc-bbb bb-wl18xx-bluetooth bb-wl18xx-wlan0"
+IMAGE_INSTALL:append_beaglebone-pocket = " boot-scripts"
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-core/images/balena-image.bbappend:4]: New syntax conversion needed here

File: layers/meta-balena-beaglebone/recipes-core/images/balena-image-flasher.bbappend [hunk 9]
```diff
@@ -1,4 +1,4 @@
 include balena-image.inc
 
-BALENA_BOOT_PARTITION_FILES_append_beaglebone = " uEnv.txt_internal:"
-IMAGE_INSTALL_append_beaglebone = " bb-org-overlays"
+BALENA_BOOT_PARTITION_FILES:append_beaglebone = " uEnv.txt_internal:"
+IMAGE_INSTALL:append_beaglebone = " bb-org-overlays"
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-core/images/balena-image-flasher.bbappend:4]: New syntax conversion needed here

File: layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot_2019.04.bbappend [hunk 10]
```diff
@@ -17,16 +18,16 @@ SRC_URI += " \
 	file://0006-Fix-boot-with-cape.patch \
 "
 
-SRC_URI_append_beaglebone-pocket = " \
+SRC_URI:append_beaglebone-pocket = " \
 	file://am335x_evm-autoboot-zero.patch \
 "
 
-SRC_URI_append_beagleboard-xm = " \
+SRC_URI:append_beagleboard-xm = " \
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot_2019.04.bbappend:25]: New syntax conversion needed here

File: layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot_2019.04.bbappend [hunk 11]
```diff
@@ -17,16 +18,16 @@ SRC_URI += " \
 	file://0006-Fix-boot-with-cape.patch \
 "
 
-SRC_URI_append_beaglebone-pocket = " \
+SRC_URI:append_beaglebone-pocket = " \
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot_2019.04.bbappend:21]: New syntax conversion needed here

File: layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging_%.bbappend [hunk 12]
```diff
@@ -3,19 +3,19 @@ inherit resin-u-boot
...
+SRC_URI:append = " \
     file://0001-Integrate-with-Balena-environment-configuration.patch \
     file://0002-am335x_evm_defconfig-Reduce-SPL-.rodata-size.patch \
     file://0005-Autoboot-keyboard-beaglebone-fixes.patch \
     file://0001-Add-support-for-BeagleBoard.org-PocketBeagle.patch \
     file://uEnv.txt_internal \
     "
 
-SRC_URI_append_beaglebone-pocket = " \
+SRC_URI:append_beaglebone-pocket = " \
```
Comment (floion) [inline layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging_%.bbappend:15]: New syntax conversion needed here

**Comment by perezmeyer** [comment]
@floion maybe some admin needs to unlock the rest of the tests?

**Comment by floion** [comment]
@resin-jenkins ok to test

**Pull request closed without merging**